### ISSUE TITLE
[1.15.2] Botania solegnolia(demagnetizer) compatibility request

### ISSUE THREAD

**Issue by theboo** [PrinceOfAmber/Cyclic#1448]
Minecraft Version: 
1.15.2
Forge Version: 
31.2.20
Mod Version: 
1.15.2-0.4.1
Single Player or Server:
both
Describe problem (what you were doing; what happened; what should have happened):

More of a mod compatibility thing than a problem. Just suggesting/requesting that the Cyclic magnet enchantment respect the botania solegnolia (their plant that keeps magnets from working in a 4 block radius around it)

**Comment by trunksbomb** [comment]
As best as I can tell, the flower doesn't actually prevent any magnets from working, rather the onus is on the magnet to detect the flower and make itself stop working.

We would need to depend on the Botania API and make a similar call to:

```
if (BotaniaAPI.instance().hasSolegnoliaAround(living)) {
```

and prevent the magnet enchantment from working if true.

**Comment by trunksbomb** [comment]
I have this fixed in dev for 1.16 by adding Botania API and the relevant method (above).

However, running the mod without Botania present causes a crash:

```
java.lang.NoClassDefFoundError: vazkii/botania/api/BotaniaAPI
```

@Lothrazar didn't you have a similar issue with Patchouli?

**Comment by trunksbomb** [comment]
Never mind, just have to detect presence of the Botania first.
